Brokerages/Zerodha/ZerodhaBrokerage.cs

```diff
@@ -0,0 +1,1373 @@
+            OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Warning, -1, message));
+
+            UnlockStream();
+            return false;
+        }
+
+        /// <summary>
+        /// Cancels the order with the specified ID
+        /// </summary>
+        /// <param name="order">The order to cancel</param>
+        /// <returns>True if the request was submitted for cancellation, false otherwise</returns>
+        public override bool CancelOrder(Order order)
+        {
+            LockStream();
+            Log.Trace("CancelOrder Order Status "+order.Status.ToString());
+            //if (order.Type == OrderType.Bracket)
+            //{
+            //    orderResponse = _kite.CancelOrder(order.Id.ToStringInvariant());
+            //}
+            //else
+            //{
+            JObject orderResponse = new JObject();
+            if (order.Status!=OrderStatus.Filled && order.Status!= OrderStatus.PartiallyFilled && order.Status!=OrderStatus.Invalid)
+            {
+                try
+                {
+                    orderResponse = _kite.CancelOrder(order.BrokerId[0].ToStringInvariant());
+                }
+                catch (Exception ex)
+                {
+                    OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Error, (string)orderResponse["status"], $"Error cancelling order: {orderResponse["status_message"]}"));
+                    UnlockStream();
+                    return false;
+                }
+            }
+            else
+            {
+                //Verify this
+                OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Error,500, $"Error cancelling open order"));
+                UnlockStream();
+                return false;
+            }
+            
+            //}
+            if ((string)orderResponse["status"] == "success")
+            {
+                OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, OrderFee.Zero,"Zerodha Order Cancelled Event") { Status = OrderStatus.Canceled });
+                UnlockStream();
+                return true;
+            }
+            var errorMessage = $"Error cancelling order: {orderResponse["status_message"]}";
+            OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Error, (string)orderResponse["status"], errorMessage));
+
+            UnlockStream();
+            return false;
+
+
+        }
+
+
+        /// <summary>
+        /// Gets all orders not yet closed
+        /// </summary>
+        /// <returns></returns>
+        public override List<Order> GetOpenOrders()
+        {
+            var allOrders = _kite.GetOrders();
+            List<Order> list = new List<Order>();
+            //Only loop if there are any actual orders inside response
+            if (allOrders.Count > 0)
+            {
+
+                foreach (var item in allOrders.Where(z => z.Status == "filled"))
```

Comment:
This method is only called once at startup to load existing active orders,
not sure why we're including filled orders here

Reply:
Fixed